components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -438,3 +450,106 @@ func InjectCertConfig(notebook *nbv1.Notebook, configMapName string) error {
+                        continue
+                    }
+
+                    // Iterate through the imagestreams to find matches
+                    for _, item := range imagestreams.Items {
+                        metadata := item.Object["metadata"].(map[string]interface{})
+                        name := metadata["name"].(string)
+
+                        if name == imagestreamName {
+                            status := item.Object["status"].(map[string]interface{})
+
+                            log.Info("No Internal registry found, pick up imageHash from status.tag.dockerImageReference")
+
+                            tags := status["tags"].([]interface{})
+                            for _, t := range tags {
+                                tagMap := t.(map[string]interface{})
+                                tagName := tagMap["tag"].(string)
+                                if tagName == tag {
+                                    items := tagMap["items"].([]interface{})
+                                    if len(items) > 0 {
+                                        // Sort items by creationTimestamp to get the most recent one
+                                        sort.Slice(items, func(i, j int) bool {
+                                            iTime := items[i].(map[string]interface{})["created"].(string)
+                                            jTime := items[j].(map[string]interface{})["created"].(string)
+                                            return iTime > jTime // Lexicographical comparison of RFC3339 timestamps
+                                        })
+                                        imageHash := items[0].(map[string]interface{})["dockerImageReference"].(string)
+                                        notebook.Spec.Template.Spec.Containers[0].Image = imageHash
+                                        // Update the JUPYTER_IMAGE environment variable
+                                        for i, envVar := range notebook.Spec.Template.Spec.Containers[0].Env {
+                                            if envVar.Name == "JUPYTER_IMAGE" {
+                                                notebook.Spec.Template.Spec.Containers[0].Env[i].Value = imageHash
+                                                break
+                                            }
+                                        }
+                                        imagestreamFound = true
+                                        break
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    if imagestreamFound {
+                        break
+                    }
+                }
+
+                if !imagestreamFound {
+                    log.Info("Imagestream not found in any of the specified namespaces", "imagestreamName", imagestreamName, "tag", tag)
+                }
+            }
+        }
```

Comment:
Would be nice from us to provide some log/warning for the else branches in these 2 cases (ifs on lines 473 and 474).